scheduler/generic_sched.go

```diff
@@ -510,8 +480,16 @@ func (s *GenericScheduler) computePlacements(destructive, place []placementResul
 
 				// If the new allocation is replacing an older allocation then we
 				// set the record the older allocation id so that they are chained
-				if prev := missing.PreviousAllocation(); prev != nil {
+				if prev := prevAllocation; prev != nil {
 					alloc.PreviousAllocation = prev.ID
+					var rescheduleEvents []*structs.RescheduleEvent
+					if prev.RescheduleTracker != nil {
+						for _, reschedEvent := range prev.RescheduleTracker.Events {
+							rescheduleEvents = append(rescheduleEvents, reschedEvent.Copy())
+						}
+					}
+					rescheduleEvents = append(rescheduleEvents, &structs.RescheduleEvent{RescheduleTime: time.Now().UTC().UnixNano(), PrevAllocID: prev.ID, PrevNodeID: alloc.NodeID})
```

Comment:
Can you pull out into a helper? Also shouldn't you only do this if there is a restart policy on the task group